scripts/library.sh

```diff
@@ -132,30 +132,47 @@ function wait_until_object_does_not_exist() {
 # Parameters: $1 - namespace.
 function wait_until_pods_running() {
   echo -n "Waiting until all pods in namespace $1 are up"
+  local failed_pod=""
   for i in {1..150}; do  # timeout after 5 minutes
     local pods="$(kubectl get pods --no-headers -n $1 2>/dev/null)"
     # All pods must be running
-    local not_running=$(echo "${pods}" | grep -v Running | grep -v Completed | wc -l)
+    local not_running_pods=$(echo "${pods}" | grep -v Running | grep -v Completed)
+    local not_running=$(echo "${not_running_pods}" | wc -l)
     if [[ -n "${pods}" && ${not_running} -eq 0 ]]; then
+      # All Pods are running or completed. Verify the containers on each Pod.
       local all_ready=1
       while read pod ; do
         local status=(`echo -n ${pod} | cut -f2 -d' ' | tr '/' ' '`)
+        local pod_name=$(echo -n ${pod} | cut -f1 -d' ')
         # All containers must be ready
-        [[ -z ${status[0]} ]] && all_ready=0 && break
-        [[ -z ${status[1]} ]] && all_ready=0 && break
-        [[ ${status[0]} -lt 1 ]] && all_ready=0 && break
-        [[ ${status[1]} -lt 1 ]] && all_ready=0 && break
-        [[ ${status[0]} -ne ${status[1]} ]] && all_ready=0 && break
+        [[ -z ${status[0]} ]] && all_ready=0 && failed_pod="${pod_name}" && break
```

Comment:
You can simplify this series of checks by setting failed_pod in line 146 only, then resetting it to empty string in line 153.